bundles/org.openhab.binding.echonetlite/src/main/java/org/openhab/binding/echonetlite/internal/EchonetChannel.java

```
@@ -0,0 +1,107 @@
+import java.nio.channels.DatagramChannel;
+import java.nio.channels.SelectionKey;
+import java.nio.channels.Selector;
+import java.util.Enumeration;
+import java.util.function.Consumer;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Wraps a Datagram channel for sending/receiving data to/from echonet lite devices.
+ *
+ * @author Michael Barker - Initial contribution
+ */
+@NonNullByDefault
+public class EchonetChannel {
+
+    private final Logger logger = LoggerFactory.getLogger(EchonetChannel.class);
+
+    private final DatagramChannel channel;
+    private final Selector selector = Selector.open();
+
+    private short tid = 0;
+
+    public EchonetChannel(InetSocketAddress discoveryAddress) throws IOException {
+        channel = DatagramChannel.open(StandardProtocolFamily.INET);
+        channel.bind(new InetSocketAddress("0.0.0.0", discoveryAddress.getPort()));
+        final Enumeration<NetworkInterface> networkInterfaces = NetworkInterface.getNetworkInterfaces();
+        while (networkInterfaces.hasMoreElements()) {
+            final NetworkInterface networkInterface = networkInterfaces.nextElement();
+            if (networkInterface.supportsMulticast() && hasIpV4Address(networkInterface)) {
+                channel.join(discoveryAddress.getAddress(), networkInterface);
+            }
+        }
+        channel.configureBlocking(false);
+        channel.register(selector, SelectionKey.OP_READ);
+    }
+
+    private boolean hasIpV4Address(final NetworkInterface networkInterface) {
+        return networkInterface.inetAddresses().anyMatch(ia -> ia instanceof Inet4Address);
+    }
+
+    public void close() {
+        try {
+            logger.debug("closing selector");
+            selector.close();
+            logger.debug("closing channel");
+            channel.close();
+        } catch (IOException e) {
+            logger.warn("Failed to close selector/channel", e);
```

Can this be silently ignored - if not, does it need to be warning?